schema-derivatives-perpfutures.graphql

```diff
@@ -0,0 +1,1425 @@
+  dailyTransactionCount: Int!
+
+  " Daily number of deposits (add liquidity) "
+  dailyDepositCount: Int!
+
+  " Daily number of withdrawals (remove liquidity) "
+  dailyWithdrawCount: Int!
+
+  " Daily number of borrows for leverage "
+  dailyBorrowCount: Int!
+
+  " Daily number of trades (swaps) "
+  dailySwapCount: Int!
+
+  " Number of unique daily depositors "
+  dailyActiveDepositors: Int!
+
+  " Number of cumulative depositors "
+  cumulativeUniqueDepositors: Int!
+
+  " Number of unique daily borrowers "
+  dailyActiveBorrowers: Int!
+
+  " Number of cumulative borrowers "
+  cumulativeUniqueBorrowers: Int!
+
+  " Number of unique daily liquidators (accounts that performed liquidation) "
+  dailyActiveLiquidators: Int!
+
+  " Number of cumulative liquidators (accounts that performed liquidation) "
+  cumulativeUniqueLiquidators: Int!
+
+  " Number of unique daily liquidatees (accounts that got liquidated) "
+  dailyActiveLiquidatees: Int!
+
+  " Number of cumulative liquidatees (accounts that got liquidated) "
+  cumulativeUniqueLiquidatees: Int!
+
+  " Total number of pools "
+  totalPoolCount: Int!
+
+
+  " Block number of this snapshot "
+  blockNumber: BigInt!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+}
+
+type UsageMetricsHourlySnapshot @entity (immutable: true) {
+  " { # of hours since Unix epoch time } "
+  id: Bytes!
+
+  " Number of hours since Unix epoch time "
+  hours: Int!
+
+  " Protocol this snapshot is associated with "
+  protocol: DerivPerpProtocol!
+
+  " Number of unique hourly active users "
+  hourlyActiveUsers: Int!
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Total number of transactions occurred in an hour. Transactions include all entities that implement the Event interface. "
+  hourlyTransactionCount: Int!
+
+  " Total number of deposits (add liquidity) in an hour "
+  hourlyDepositCount: Int!
+
+  " Total number of withdrawals (remove liquidity) in an hour "
+  hourlyWithdrawCount: Int!
+
+  " Total number of trades (swaps) in an hour "
+  hourlySwapCount: Int!
+
+  " Block number of this snapshot "
+  blockNumber: BigInt!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+}
+
+type FinancialsDailySnapshot @entity (immutable: true) {
+  " ID is # of days since Unix epoch time "
+  id: Bytes!
+
+  " Number of days since Unix epoch time "
+  days: Int!
+
+  " Protocol this snapshot is associated with "
+  protocol: DerivPerpProtocol!
+
+  " Current TVL (Total Value Locked) of the entire protocol "
+  totalValueLockedUSD: BigDecimal!
+
+  " Daily trade Notional volume, in USD "
+  dailyVolumeUSD: BigDecimal!
+
+  " All historical trade Notional volume in USD "
+  cumulativeVolumeUSD: BigDecimal!
+  
+  " Daily Inflow trade Notional volume, in USD, based on value of underlying assets coming into protocol from trade "
+  dailyInflowVolumeUSD: BigDecimal!
```

Comment:
How should we count this? Once a position is closed? Or whenever they are opened?

Reply:
Plan: Have two metrics: Open inflow (`dailyInflowVolumeUSD`) and closed inflow (`dailyClosedInflowVolumeUSD`)

Two derived metrics `dailyInflowVolumeUSD`-`dailyOutflowVolumeUSD` for general flow
`dailyClosedInflowVolumeUSD`-`dailyOutflowVolumeUSD` for profitability